### DISCUSSION TITLE
GluonTS with intermittent data

### DISCUSSION THREAD

**Discussion by sarah-almaghrabi** [awslabs/gluonts#1831] [Q&A]
I've just started using Gluon as I need it to evaluate the built-in models on a custom dataset. 

The dataset that I have represents intermittent time series. It is for hourly observations for the period from 6:00 am to 7:00 pm.  What is the proper way to represent this data using Gluon?. 

As I understand we can use the following statement to prepare the data with 1h freq

```
        training_data = ListDataset(
        [{"start": df.index[0], "target": df.pv[:"1958-12-01], }],
        freq = "1H"
    )
```

However, the data that I have does not represent the 24 hours but only 14 observations per day so what is the proper frequency to have? 

I did try to reshape the data into the (days, hours) shape, and use the following statement. 
    
```

    training_data = ListDataset(
            [{"start": df.index[0], "target": df[:"1958-12-01], }],
            freq = "1D"
        )
```

But I got this error: 

`    GluonTSDataError: Array 'target' has bad shape - expected 1 dimensions, got 2.
`

Any suggestions are highly appreciated, 
Thanks

**Comment by rsnirwan** [comment]
I am not sure about the intermittent data but the error message seems to results from a different source.
The "target" needs to be a one dimensional object (e.g. list of numbers). In you case it is a matrix.

If you have multiple time series, you need a dict for each: 
ListDataset([{"start": "..", "target": ts1}, {"start": "..", "target": ts2}], freq="1H")
